OpenEdXMobile/src/main/java/org/edx/mobile/module/analytics/SegmentAnalytics.java

```diff
@@ -68,49 +70,111 @@ private void setCustomProperties() {
             this.properties.putValue(Keys.NAVIGATION_MODE,
                     (isVideoMode ? Values.OUTLINE_MODE_VIDEO : Values.OUTLINE_MODE_FULL));
         }
-    }
 
-    /**
-     * This function is set to identify the user for subsequent calls
-     *
-     * @param userID   - User Id from the server
-     * @param email    -  email of the user
-     * @param username -  Username/email that the user uses for signing in
-     * @return A {@link Traits} object populated with provided user info
-     */
-    @Override
-    public Traits identifyUser(String userID, String email, String username) {
-        Traits traits = new Traits();
-        traits.putEmail(email);
-        traits.putUsername(username);
-        tracker.identify(userID, traits, new Options());
-        return traits;
+        /**
+         * This function sets the Context values of values passed
+         *
+         * @param courseId
+         * @param unitUrl
+         * @param component
+         * @return A {@link Properties} object populated with analytics-event info
+         */
+        private Properties getEventContext(String courseId, String unitUrl, String component) {
+            Properties cxtProps = new Properties();
+            if (courseId != null) {
+                cxtProps.putValue(Keys.COURSE_ID, courseId);
+            }
+            if (unitUrl != null) {
+                cxtProps.putValue(Keys.OPEN_BROWSER, unitUrl);
+            }
+            if (component != null) {
+                cxtProps.putValue(Keys.COMPONENT, component);
+            }
+            cxtProps.putValue(Keys.APP, Values.APP_NAME);
+
+            return cxtProps;
+        }
     }
 
     /**
-     * This function is used to send the screen tracking event.
-     *
-     * @param screenName The screen name to track
-     * @return A {@link Properties} object populated with analytics-event info
+     * Utility class that defines the basic structure for a Segment tracker that we can use for
+     * sending analytics events.
      */
-    @Override
-    public Properties trackScreenView(@NonNull String screenName) {
-        return trackScreenView(screenName, null, null);
-    }
+    private static class SegmentTracker {
```

Comment:
This class seems redundant though, since it's only purpose seems to be to ensure null checking on the `analytics` instance, which should never be null in any case (since the `isEnabled()` method also verifies a non-null write key).